src/GameLogic.cpp

```diff
@@ -102,6 +56,28 @@ void GameLogic::update(float timeS) {
 void GameLogic::setLevel(int level) {
     currentLevel = level;
     //load specified level
+    loader.LoadLevel(level);
+    platVector = loader.getPlatforms();
+    spikeVector = loader.getSpikes();
+    enemyVector = loader.getEnemies();
```

Comment:
I would reduce the scope of these variables to being declared just in this function since they are not being used anywhere else in GameLogic and do not need to be member variables

Reply:
Doing this is causing a seg fault, and when I declare them locally with a specific size I am getting a Bus Error 10. I don't think I understand enough about C++ to know why this might be happening, any suggestions?

Reply:
They cannot be declared locally because then they go out of scope as soon as that local scope ends and they get destroyed. This is the easiest way around that without having to start allocating memory space on the heap which opens up a whole host of problems with memory management and memory leaks. It is also the reason why I defined them like this. I am pretty sure that we still need the vectors of reference_wrappers though. I am going to go ahead and empty them after we use them in setLevel though, so that they are not taking up space in memory.

Reply:
Actually, now that I think about it, I think I can change it so that platforms, spikes, and enemies are normal vectors and only actorsVector is a vector of reference_wrappers, since actorsVector is the only one that gets passed out of GameLogic to HumanView. I will go ahead and work on that.